pkg/apis/kubeone/config/config.go

```diff
@@ -214,26 +234,30 @@ func DefaultedV1Beta3KubeOneCluster(versionedCluster *kubeonev1beta3.KubeOneClus
 		return nil, fail.Config(err, fmt.Sprintf("converting %s to internal object", versionedCluster.GroupVersionKind()))
 	}
 
-	// Apply the dynamic defaults
-	if err := SetKubeOneClusterDynamicDefaults(internalCluster, credentialsFile); err != nil {
-		return nil, err
-	}
-
-	// Validate the configuration
-	if err := kubeonevalidation.ValidateKubeOneCluster(*internalCluster).ToAggregate(); err != nil {
-		return nil, fail.ConfigValidation(err)
-	}
-
-	// Check for deprecated fields/features for a cluster
-	checkClusterConfiguration(*internalCluster, logger)
-
 	return internalCluster, nil
 }
 
 // SetKubeOneClusterDynamicDefaults sets the dynamic defaults for a given KubeOneCluster object
-func SetKubeOneClusterDynamicDefaults(cluster *kubeoneapi.KubeOneCluster, credentialsFile []byte) error {
+func SetKubeOneClusterDynamicDefaults(cluster *kubeoneapi.KubeOneCluster, credentialsFile []byte, baseDir string) error {
 	// Set the default cloud config
 	SetDefaultsCloudConfig(cluster)
+
+	if cluster.CertificateAuthority.File != "" && !filepath.IsAbs(cluster.CertificateAuthority.File) {
+		cluster.CertificateAuthority.File = filepath.Join(baseDir, cluster.CertificateAuthority.File)
+	}
+
+	if cluster.CertificateAuthority.File != "" {
+		buf, err := os.ReadFile(cluster.CertificateAuthority.File)
+		if err != nil {
+			return fail.ConfigValidation(err)
+		}
+
+		cluster.CertificateAuthority.Bundle = string(buf)
+	}
+
+	// Set this for backward compatibility with older addons
+	cluster.CABundle = cluster.CertificateAuthority.Bundle
```

Comment:
Isn't this going to fail the validation later on?

Reply:
So I've removed that validation, replaced with defaulting that will overwrite caBundle in case when certificateAuthority.bundle is set, and set certificateAuthority.bundle (if it's empty) as copy of caBundle.